### ISSUE TITLE
CMake cannot verify NVCC compiler after rerender to 2019 Windows Image

### ISSUE THREAD

**Issue by carterbox** [conda-forge/nvcc-feedstock#76]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:
After a feedstock was [re-rendered to the 2019 Windows image](https://github.com/conda-forge/libastra-feedstock/pull/7), CMake can no longer verify NVCC as a working compiler. The version of the nvcc_win-64 is the same before and after the rerender. At first, I thought the problem was that the wrong host compiler was being assigned to NVCC because the version of the compiler linked to %CXX% did not match the host compiler reported by CMake, but then I also tried manually assigning NVCC and host compiler paths.

I really don't know what's going on.

<br/>
Environment (<code>conda list</code>):
<details>

```
    cmake:          3.21.3-h39d44d4_0       conda-forge
    ninja:          1.10.2-h2d74725_1       conda-forge
    nvcc_win-64:    11.2-h929061a_15        conda-forge
    ucrt:           10.0.20348.0-h57928b3_0 conda-forge
    vc:             14.2-hb210afc_5         conda-forge
    vs2015_runtime: 14.29.30037-h902a5da_5  conda-forge
    vs2017_win-64:  19.16.27038-h2e3bad8_2  conda-forge
    vswhere:        2.8.4-h57928b3_0        conda-forge


```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

**Comment by leofang** [comment]
cc: @jaimergp for vis

**Comment by leofang** [comment]
Looks like we have a growing number of people hitting this issue:
```
'cl.exe' is not recognized as an internal or external command,
operable program or batch file.
```
- https://github.com/conda-forge/libastra-feedstock/pull/7 (@carterbox)
- https://github.com/conda-forge/pyvkfft-feedstock/pull/4 (@isuruf)
- https://github.com/conda-forge/cupy-feedstock/pull/173 (@leofang)

Sounds like vs2019 is to blame?

cc: @conda-forge/core @conda-forge/nvcc @conda-forge/cudatoolkit

**Comment by leofang** [comment]
I figured it out. We need to set `--use-local-env` flag to nvcc. Let me send a patch.

**Comment by leofang** [comment]
I could be wrong, but I suspect we never caught this bug in this feedstock because `nvcc.bat` is never used. From the Windows docs
> The PATHEXT environment variable defines the list of file extensions checked by Windows NT when searching for an executable file. Like the PATH variable, semi-colons separate individual items in the PATHEXT variable. The default value of PATHEXT is .COM;.EXE;.BAT;.CMD. 

So it might be possible that we've been using `nvcc.exe` from CTK instead of `nvcc.bat` all the time.

**Comment by leofang** [comment]
https://github.com/conda-forge/nvcc-feedstock/pull/88#issuecomment-1172916314

**Comment by hadim** [comment]
I am not sure what is the status here but also got the `cl.exe` error at https://github.com/conda-forge/katago-feedstock/pull/1 and when trying to use your fix:

```
    set CUDNN_INCLUDE_DIR=%LIBRARY_PREFIX%\include
    for /f "tokens=* usebackq" %%f in (`where nvcc`) do (set "dummy=%%f" && call set "NVCC=%%dummy:\=\\%%")
    set "NVCC=%NVCC% --use-local-env"
    echo "nvcc is %NVCC%, CUDA path is %CUDA_PATH%"
```

then I got a `Failed to find nvcc.` error.

Any idea?

**Comment by leofang** [comment]
No, unfortunately. I don't really know what's going on. The patch #88 did not work...

**Comment by jaimergp** [comment]
[Isuru's patch did work in this other PR,](https://github.com/conda-forge/pyvkfft-feedstock/pull/4/commits/7bd7b0f29e10ca0cd9f143561011ea6fbfc86d27) so I don't know :(

**Comment by leofang** [comment]
Interesting, @isuruf have you figured out what's missing in #88?
